python/cuml/preprocessing/model_selection.py

```diff
@@ -121,21 +117,27 @@ def train_test_split(
     # todo: this check will be replaced with upcoming improvements
     # to input_utils
     #
-    if not hasattr(X, "__cuda_array_interface__") and not \
-            isinstance(X, cudf.DataFrame) and isinstance(y, cudf.Series):
-        raise TypeError("X needs to be either a cuDF DataFrame, Series or \
-                        a cuda_array_interface compliant array.")
-
-    if not hasattr(y, "__cuda_array_interface__") and not \
-            isinstance(y, cudf.DataFrame) and isinstance(y, cudf.Series):
-        raise TypeError("y needs to be either a cuDF DataFrame, Series or \
-                        a cuda_array_interface compliant array.")
-
-    if X.shape[0] != y.shape[0]:
-        raise ValueError(
-            "X and y must have the same first dimension"
-            "(found {} and {})".format(X.shape[0], y.shape[0])
-        )
+    if y is not None:
+        if not hasattr(X, "__cuda_array_interface__") and not \
+                isinstance(X, cudf.DataFrame) and isinstance(y, cudf.Series):
```

Comment:
```suggestion
                isinstance(X, cudf.DataFrame):
```
cuDF Series are cuda array interface compliant, so no need to check them explicitly